modules/auxiliary/scanner/redis/redis_login.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'metasploit/framework/login_scanner/redis'
+require 'metasploit/framework/credential_collection'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'         => 'Redis Login Utility',
+        'Description'  => 'This module attempts to authenticate to an REDIS service.',
+        'Author'       => [ 'Nixawk' ],
+        'References'   => [
+          ['URL', 'http://redis.io/topics/protocol']
+        ],
+        'License'      => MSF_LICENSE))
+
+    register_options(
+      [
+        Opt::RPORT(6379),
```

Comment:
In this module you should be able to import the redis mixin as is and some of this code will be simplified.